src/checkbox/checkbox.js

```diff
@@ -56,6 +56,40 @@ class MaterialCheckbox extends MaterialComponent {
   }
 
   /**
+   * Generates an ID for a new checkbox. Keeps a counter to ensure uniqueness.
+   * @return {string} The generated ID.
+   */
+  static generateId_() {
+    MaterialCheckbox.autoCounter_ = (MaterialCheckbox.autoCounter_ || 0) + 1;
+    return `mdlcheckbox-${MaterialCheckbox.autoCounter_}`;
+  }
+
+  /**
+   * Creates the DOM subtree for a new component.
+   * Greatly simplifies programmatic component creation.
+   *
+   * @protected
+   * @nocollapse
+   * @return {Element} The DOM subtree for the component.
+   */
+  static buildDom_() {
+    let root = document.createElement('label');
+    let input = document.createElement('input');
+    let label = document.createElement('span');
+    let id = MaterialCheckbox.generateId_();
```

Comment:
We can still dump this auto-generation and everything attached to it. The input is a child of the label element so it is inherent that the relation exists.